util/dvsim/FlowCfg.py

```diff
@@ -470,182 +466,128 @@ def _get_results_page_link(self, relative_to, link_text=''):
+                                             self.results_html_name)
+            history_txt += '- [{}]({})\n'.format(existing_basename,
+                                                 relative_url)
 
         # Append the history to the results.
         publish_results_md = self.publish_results_md or self.results_md
         publish_results_md = publish_results_md + history_txt
 
-        # Publish the results page.
-        # First, write the results html and json files to the scratch area.
-        json_str = (json.dumps(self.results_dict)
-                    if hasattr(self, 'results_dict')
-                    else None)
+        # Export any results dictionary to json
+        suffixes = ['html']
+        json_str = None
+        if hasattr(self, 'results_dict'):
+            suffixes.append('json')
+            json_str = json.dumps(self.results_dict)
+
+        # Export our markdown page to HTML and dump the json to a local file.
+        # These are called publish.html and publish.json locally, but we'll
+        # rename them as part of the upload.
         self.write_results("publish.html", publish_results_md, json_str)
-        results_html_file = self.results_dir / "publish.html"
 
-        # Second, copy the files to the server.
+        html_name_no_suffix = self.results_html_name.split('.', 1)[0]
+        dst_no_suffix = '{}/latest/{}'.format(self.rel_path,
+                                              html_name_no_suffix)
+
+        # Now copy our local files over to the server
         log.info("Publishing results to %s", self.results_server_url)
-        suffixes = ['html'] + (['json'] if json_str is not None else [])
         for suffix in suffixes:
-            src = str(Path(results_html_file).with_suffix('.' + suffix))
-            dst = self.results_server_page
-            # results_server_page has '.html' as suffix.  If that does not match
-            # suffix, change it.
-            if suffix != 'html':
-                assert dst[-5:] == '.html'
-                dst = dst[:-5] + '.json'
-            cmd = f"{self.results_server_cmd} cp {src} {dst}"
-            log.log(VERBOSE, cmd)
-            try:
-                cmd_output = subprocess.run(args=cmd,
-                                            shell=True,
-                                            stdout=subprocess.PIPE,
-                                            stderr=subprocess.STDOUT)
-                log.log(VERBOSE, cmd_output.stdout.decode("utf-8"))
-            except Exception as e:
-                log.error("%s: Failed to publish results:\n\"%s\"", e, str(cmd))
+            src = "{}/publish.{}".format(self.results_dir, suffix)
+            dst = "{}.{}".format(dst_no_suffix, suffix)
+            results_server.upload(src, dst)
 
     def publish_results(self):
         '''Public facing API for publishing results to the opentitan web
         server.
         '''
+        try:
+            results_server = ResultsServer(self.results_server)
```

Comment:
Could one of these variables get a different name?

Reply:
Changing the field's name would be fiddly (because I think Python names and hjson names get conflated). So I've changed the name here to make the code look less odd, but I've left the the names for the arguments it becomes as "results_server" because I think that's clearer.

I guess another option would be to rename the class to something like "ResultsServerHandle", but that feels a bit lame so I haven't done it yet.